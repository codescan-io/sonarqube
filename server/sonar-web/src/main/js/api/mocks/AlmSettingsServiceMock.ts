/*
 * SonarQube
 * Copyright (C) 2009-2023 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { cloneDeep } from 'lodash';
import { mockAlmSettingsInstance } from '../../helpers/mocks/alm-settings';
import { AlmKeys, AlmSettingsInstance } from '../../types/alm-settings';
import { getAlmSettings } from '../alm-settings';

export default class AlmSettingsServiceMock {
  almSettings: AlmSettingsInstance[];
  defaultSetting: AlmSettingsInstance[] = [
    mockAlmSettingsInstance({ key: 'conf-final-1', alm: AlmKeys.GitLab }),
    mockAlmSettingsInstance({ key: 'conf-final-2', alm: AlmKeys.GitLab }),
    mockAlmSettingsInstance({ key: 'conf-github-1', alm: AlmKeys.GitHub, url: 'http://url' }),
    mockAlmSettingsInstance({ key: 'conf-github-2', alm: AlmKeys.GitHub, url: 'http://url' }),
    mockAlmSettingsInstance({ key: 'conf-github-3', alm: AlmKeys.GitHub, url: 'javascript://url' }),
    mockAlmSettingsInstance({ key: 'conf-azure-1', alm: AlmKeys.Azure, url: 'url' }),
    mockAlmSettingsInstance({ key: 'conf-azure-2', alm: AlmKeys.Azure, url: 'url' }),
    mockAlmSettingsInstance({
      key: 'conf-bitbucketcloud-1',
      alm: AlmKeys.BitbucketCloud,
      url: 'url',
    }),
    mockAlmSettingsInstance({
      key: 'conf-bitbucketcloud-2',
      alm: AlmKeys.BitbucketCloud,
      url: 'url',
    }),
    mockAlmSettingsInstance({
      key: 'conf-bitbucketserver-1',
      alm: AlmKeys.BitbucketServer,
      url: 'url',
    }),
    mockAlmSettingsInstance({
      key: 'conf-bitbucketserver-2',
      alm: AlmKeys.BitbucketServer,
      url: 'url',
    }),
  ];

  constructor() {
    this.almSettings = cloneDeep(this.defaultSetting);
    jest.mocked(getAlmSettings).mockImplementation(this.getAlmSettingsHandler);
  }

  getAlmSettingsHandler = () => {
    return Promise.resolve(this.almSettings);
  };

  reset = () => {
    this.almSettings = cloneDeep(this.defaultSetting);
  };
}
