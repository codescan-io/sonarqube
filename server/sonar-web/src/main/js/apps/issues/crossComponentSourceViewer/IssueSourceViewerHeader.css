/*
 * SonarQube
 * Copyright (C) 2009-2023 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
.issue-source-viewer-header {
  padding: 4px 10px;
  border: 1px solid var(--gray80);
  background-color: var(--barBackgroundColor);
  align-items: center;
  min-height: 25px;
  position: sticky;
  z-index: 100;
  top: 0;
  margin-top: 8px;
  margin-bottom: -1px;
}

.issue-source-viewer-header:first-child {
  margin-top: 0;
}
