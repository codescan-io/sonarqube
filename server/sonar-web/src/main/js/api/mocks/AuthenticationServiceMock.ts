/*
 * SonarQube
 * Copyright (C) 2009-2023 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { cloneDeep } from 'lodash';
import { mockSettingValue } from '../../helpers/mocks/settings';
import { BranchParameters } from '../../types/branch-like';
import { SettingDefinition, SettingValue } from '../../types/settings';
import { getValues, resetSettingValue, setSettingValue } from '../settings';

export default class AuthenticationServiceMock {
  settingValues: SettingValue[];
  defaulSettingValues: SettingValue[] = [
    mockSettingValue({ key: 'test1', value: '' }),
    mockSettingValue({ key: 'test2', value: 'test2' }),
    mockSettingValue({ key: 'sonar.auth.saml.certificate.secured' }),
    mockSettingValue({ key: 'sonar.auth.saml.enabled', value: 'false' }),
  ];

  constructor() {
    this.settingValues = cloneDeep(this.defaulSettingValues);
    (getValues as jest.Mock).mockImplementation(this.getValuesHandler);
    (setSettingValue as jest.Mock).mockImplementation(this.setValueHandler);
    (resetSettingValue as jest.Mock).mockImplementation(this.resetValueHandler);
  }

  getValuesHandler = (data: { keys: string; component?: string } & BranchParameters) => {
    if (data.keys) {
      return Promise.resolve(this.settingValues.filter((set) => data.keys.includes(set.key)));
    }
    return Promise.resolve(this.settingValues);
  };

  setValueHandler = (definition: SettingDefinition, value: string) => {
    if (value === 'error') {
      const res = new Response('', {
        status: 400,
        statusText: 'fail',
      });

      return Promise.reject(res);
    }
    const updatedSettingValue = this.settingValues.find((set) => set.key === definition.key);
    if (updatedSettingValue) {
      updatedSettingValue.value = value;
    }
    return Promise.resolve();
  };

  resetValueHandler = (data: { keys: string; component?: string } & BranchParameters) => {
    if (data.keys) {
      this.settingValues.forEach((set) => {
        if (data.keys.includes(set.key)) {
          set.value = '';
        }
        return set;
      });
    }
    return Promise.resolve();
  };

  resetValues = () => {
    this.settingValues = cloneDeep(this.defaulSettingValues);
  };
}
