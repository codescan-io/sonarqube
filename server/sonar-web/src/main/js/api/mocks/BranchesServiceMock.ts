/*
 * SonarQube
 * Copyright (C) 2009-2023 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { cloneDeep } from 'lodash';
import { mockBranch } from '../../helpers/mocks/branch-like';
import { BranchLike } from '../../types/branch-like';
import { getBranches } from '../branches';

export default class BranchesServiceMock {
  branchLikes: BranchLike[];
  defaultBranchLikes: BranchLike[] = [
    mockBranch({ isMain: true, name: 'master' }),
    mockBranch({ excludedFromPurge: false, name: 'delete-branch' }),
    mockBranch({ name: 'normal-branch' }),
  ];

  constructor() {
    this.branchLikes = cloneDeep(this.defaultBranchLikes);
    (getBranches as jest.Mock).mockImplementation(this.getBranchesHandler);
  }

  getBranchesHandler = () => {
    return Promise.resolve(this.branchLikes);
  };

  resetBranches = () => {
    this.branchLikes = cloneDeep(this.defaultBranchLikes);
  };
}
