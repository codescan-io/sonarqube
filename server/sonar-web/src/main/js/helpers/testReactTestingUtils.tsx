/*
 * SonarQube
 * Copyright (C) 2009-2023 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { render, RenderResult } from '@testing-library/react';
import * as React from 'react';
import { HelmetProvider } from 'react-helmet-async';
import { IntlProvider } from 'react-intl';
import { MemoryRouter, Outlet, parsePath, Route, Routes } from 'react-router-dom';
import AdminContext from '../app/components/AdminContext';
import AppStateContextProvider from '../app/components/app-state/AppStateContextProvider';
import { AvailableFeaturesContext } from '../app/components/available-features/AvailableFeaturesContext';
import { ComponentContext } from '../app/components/componentContext/ComponentContext';
import CurrentUserContextProvider from '../app/components/current-user/CurrentUserContextProvider';
import GlobalMessagesContainer from '../app/components/GlobalMessagesContainer';
import IndexationContextProvider from '../app/components/indexation/IndexationContextProvider';
import { LanguagesContext } from '../app/components/languages/LanguagesContext';
import { MetricsContext } from '../app/components/metrics/MetricsContext';
import { useLocation } from '../components/hoc/withRouter';
import { AppState } from '../types/appstate';
import { ComponentContextShape } from '../types/component';
import { Feature } from '../types/features';
import { Dict, Extension, Languages, Metric, SysStatus } from '../types/types';
import { CurrentUser } from '../types/users';
import { DEFAULT_METRICS } from './mocks/metrics';
import { mockAppState, mockCurrentUser } from './testMocks';

export interface RenderContext {
  metrics?: Dict<Metric>;
  appState?: AppState;
  languages?: Languages;
  currentUser?: CurrentUser;
  navigateTo?: string;
  featureList?: Feature[];
}

export function renderAppWithAdminContext(
  indexPath: string,
  routes: () => JSX.Element,
  context: RenderContext = {},
  overrides: { systemStatus?: SysStatus; adminPages?: Extension[] } = {}
): RenderResult {
  function MockAdminContainer() {
    return (
      <AdminContext.Provider
        value={{
          fetchSystemStatus: () => {
            /*noop*/
          },
          fetchPendingPlugins: () => {
            /*noop*/
          },
          pendingPlugins: { installing: [], removing: [], updating: [] },
          systemStatus: overrides.systemStatus ?? 'UP',
        }}
      >
        <Outlet
          context={{
            adminPages: overrides.adminPages ?? [],
          }}
        />
      </AdminContext.Provider>
    );
  }

  return renderRoutedApp(
    <Route element={<MockAdminContainer />} path="admin">
      {routes()}
    </Route>,
    indexPath,
    context
  );
}

export function renderComponent(component: React.ReactElement, pathname = '/') {
  function Wrapper({ children }: { children: React.ReactElement }) {
    return (
      <IntlProvider defaultLocale="en" locale="en">
        <MemoryRouter initialEntries={[pathname]}>
          <Routes>
            <Route path="*" element={children} />
          </Routes>
        </MemoryRouter>
      </IntlProvider>
    );
  }

  return render(component, { wrapper: Wrapper });
}

export function renderAppWithComponentContext(
  indexPath: string,
  routes: () => JSX.Element,
  context: RenderContext = {},
  componentContext?: Partial<ComponentContextShape>
) {
  function MockComponentContainer() {
    return (
      <ComponentContext.Provider
        value={{
          branchLikes: [],
          onBranchesChange: jest.fn(),
          onComponentChange: jest.fn(),
          ...componentContext,
        }}
      >
        <Outlet />
      </ComponentContext.Provider>
    );
  }

  return renderRoutedApp(
    <Route element={<MockComponentContainer />}>{routes()}</Route>,
    indexPath,
    context
  );
}

export function renderApp(
  indexPath: string,
  component: JSX.Element,
  context: RenderContext = {}
): RenderResult {
  return renderRoutedApp(<Route path={indexPath} element={component} />, indexPath, context);
}

export function renderAppRoutes(
  indexPath: string,
  routes: () => JSX.Element,
  context?: RenderContext
): RenderResult {
  return renderRoutedApp(routes(), indexPath, context);
}

export function CatchAll() {
  const location = useLocation();

  return <div>{`${location.pathname}${location.search}`}</div>;
}

function renderRoutedApp(
  children: React.ReactElement,
  indexPath: string,
  {
    currentUser = mockCurrentUser(),
    navigateTo = indexPath,
    metrics = DEFAULT_METRICS,
    appState = mockAppState(),
    featureList = [],
    languages = {},
  }: RenderContext = {}
): RenderResult {
  const path = parsePath(navigateTo);
  path.pathname = `/${path.pathname}`;

  return render(
    <HelmetProvider context={{}}>
      <IntlProvider defaultLocale="en" locale="en">
        <MetricsContext.Provider value={metrics}>
          <LanguagesContext.Provider value={languages}>
            <AvailableFeaturesContext.Provider value={featureList}>
              <CurrentUserContextProvider currentUser={currentUser}>
                <AppStateContextProvider appState={appState}>
                  <IndexationContextProvider>
                    <GlobalMessagesContainer />
                    <MemoryRouter initialEntries={[path]}>
                      <Routes>
                        {children}
                        <Route path="*" element={<CatchAll />} />
                      </Routes>
                    </MemoryRouter>
                  </IndexationContextProvider>
                </AppStateContextProvider>
              </CurrentUserContextProvider>
            </AvailableFeaturesContext.Provider>
          </LanguagesContext.Provider>
        </MetricsContext.Provider>
      </IntlProvider>
    </HelmetProvider>
  );
}
