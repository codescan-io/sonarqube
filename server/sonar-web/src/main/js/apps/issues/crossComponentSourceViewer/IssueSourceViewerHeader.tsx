/*
 * SonarQube
 * Copyright (C) 2009-2023 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import classNames from 'classnames';
import * as React from 'react';
import Link from '../../../components/common/Link';
import { ButtonIcon } from '../../../components/controls/buttons';
import { ClipboardIconButton } from '../../../components/controls/clipboard';
import ExpandSnippetIcon from '../../../components/icons/ExpandSnippetIcon';
import QualifierIcon from '../../../components/icons/QualifierIcon';
import DeferredSpinner from '../../../components/ui/DeferredSpinner';
import { getBranchLikeQuery } from '../../../helpers/branch-like';
import { translate } from '../../../helpers/l10n';
import { collapsedDirFromPath, fileFromPath } from '../../../helpers/path';
import { getBranchLikeUrl, getComponentIssuesUrl, getPathUrlAsString } from '../../../helpers/urls';
import { BranchLike } from '../../../types/branch-like';
import { ComponentQualifier } from '../../../types/component';
import { SourceViewerFile } from '../../../types/types';
import './IssueSourceViewerHeader.css';

export interface Props {
  branchLike: BranchLike | undefined;
  className?: string;
  expandable?: boolean;
  displayProjectName?: boolean;
  linkToProject?: boolean;
  loading?: boolean;
  onExpand?: () => void;
  sourceViewerFile: SourceViewerFile;
}

export default function IssueSourceViewerHeader(props: Props) {
  const {
    branchLike,
    className,
    expandable,
    displayProjectName = true,
    linkToProject = true,
    loading,
    onExpand,
    sourceViewerFile,
  } = props;
  const { measures, path, project, projectName, q } = sourceViewerFile;

  const projectNameLabel = (
    <>
      <QualifierIcon qualifier={ComponentQualifier.Project} /> <span>{projectName}</span>
    </>
  );

  const isProjectRoot = q === ComponentQualifier.Project;

  return (
    <div
      className={classNames(
        'issue-source-viewer-header display-flex-row display-flex-space-between',
        className
      )}
      role="separator"
      aria-label={sourceViewerFile.path}
    >
      <div className="display-flex-center flex-1">
        {displayProjectName && (
          <div className="spacer-right">
            {linkToProject ? (
              <a
                className="link-no-underline"
                href={getPathUrlAsString(getBranchLikeUrl(project, branchLike))}
              >
                {projectNameLabel}
              </a>
            ) : (
              projectNameLabel
            )}
          </div>
        )}

        {!isProjectRoot && (
          <>
            <div className="spacer-right">
              <QualifierIcon qualifier={q} /> <span>{collapsedDirFromPath(path)}</span>
              <span className="component-name-file">{fileFromPath(path)}</span>
            </div>

            <div className="spacer-right">
              <ClipboardIconButton
                className="button-link link-no-underline"
                copyValue={path}
                aria-label={translate('source_viewer.click_to_copy_filepath')}
              />
            </div>
          </>
        )}
      </div>

      {!isProjectRoot && measures.issues !== undefined && (
        <div
          className={classNames('flex-0 big-spacer-left', {
            'little-spacer-right': !expandable || loading,
          })}
        >
          <Link
            to={getComponentIssuesUrl(project, {
              ...getBranchLikeQuery(branchLike),
              files: path,
              resolved: 'false',
            })}
          >
            {translate('source_viewer.view_all_issues')}
          </Link>
        </div>
      )}

      {expandable && (
        <DeferredSpinner className="little-spacer-right" loading={loading}>
          <div className="flex-0 big-spacer-left">
            <ButtonIcon
              aria-label={translate('source_viewer.expand_all_lines')}
              className="js-actions"
              onClick={onExpand}
            >
              <ExpandSnippetIcon />
            </ButtonIcon>
          </div>
        </DeferredSpinner>
      )}
    </div>
  );
}
