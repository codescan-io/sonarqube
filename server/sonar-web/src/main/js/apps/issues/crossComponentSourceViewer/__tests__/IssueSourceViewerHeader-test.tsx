/*
 * SonarQube
 * Copyright (C) 2009-2023 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { mockMainBranch } from '../../../../helpers/mocks/branch-like';
import { mockSourceViewerFile } from '../../../../helpers/mocks/sources';
import { renderComponent } from '../../../../helpers/testReactTestingUtils';
import { byRole, byText } from '../../../../helpers/testSelector';
import IssueSourceViewerHeader, { Props } from '../IssueSourceViewerHeader';

const ui = {
  expandAllLines: byRole('button', { name: 'source_viewer.expand_all_lines' }),
  projectLink: byRole('link', { name: 'MyProject' }),
  projectName: byText('MyProject'),
  viewAllIssues: byRole('link', { name: 'source_viewer.view_all_issues' }),
};

it('should render correctly', () => {
  renderIssueSourceViewerHeader();

  expect(ui.expandAllLines.get()).toBeInTheDocument();
  expect(ui.projectLink.get()).toBeInTheDocument();
  expect(ui.projectName.get()).toBeInTheDocument();
  expect(ui.viewAllIssues.get()).toBeInTheDocument();
});

it('should not render expandable link', () => {
  renderIssueSourceViewerHeader({ expandable: false });

  expect(ui.expandAllLines.query()).not.toBeInTheDocument();
});

it('should not render link to project', () => {
  renderIssueSourceViewerHeader({ linkToProject: false });

  expect(ui.projectLink.query()).not.toBeInTheDocument();
  expect(ui.projectName.get()).toBeInTheDocument();
});

it('should not render project name', () => {
  renderIssueSourceViewerHeader({ displayProjectName: false });

  expect(ui.projectLink.query()).not.toBeInTheDocument();
  expect(ui.projectName.query()).not.toBeInTheDocument();
});

it('should render without issue expand all when no issue', () => {
  renderIssueSourceViewerHeader({
    sourceViewerFile: mockSourceViewerFile('foo/bar.ts', 'my-project', {
      measures: {},
    }),
  });

  expect(ui.viewAllIssues.query()).not.toBeInTheDocument();
});

function renderIssueSourceViewerHeader(props: Partial<Props> = {}) {
  return renderComponent(
    <IssueSourceViewerHeader
      branchLike={mockMainBranch()}
      expandable
      onExpand={jest.fn()}
      sourceViewerFile={mockSourceViewerFile('foo/bar.ts', 'my-project')}
      {...props}
    />
  );
}
