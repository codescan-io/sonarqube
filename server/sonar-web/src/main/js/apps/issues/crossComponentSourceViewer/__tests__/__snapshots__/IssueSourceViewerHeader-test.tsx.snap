// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  aria-label="foo/bar.ts"
  className="issue-source-viewer-header display-flex-row display-flex-space-between"
  role="separator"
>
  <div
    className="display-flex-center flex-1"
  >
    <div
      className="spacer-right"
    >
      <a
        className="link-no-underline"
        href="/dashboard?id=my-project"
      >
        <QualifierIcon
          qualifier="TRK"
        />
         
        <span>
          MyProject
        </span>
      </a>
    </div>
    <div
      className="spacer-right"
    >
      <QualifierIcon
        qualifier="FIL"
      />
       
      <span>
        foo/
      </span>
      <span
        className="component-name-file"
      >
        bar.ts
      </span>
    </div>
    <div
      className="spacer-right"
    >
      <ClipboardIconButton
        aria-label="source_viewer.click_to_copy_filepath"
        className="button-link link-no-underline"
        copyValue="foo/bar.ts"
      />
    </div>
  </div>
  <div
    className="flex-0 big-spacer-left"
  >
    <ForwardRef(Link)
      to={
        {
          "hash": "",
          "pathname": "/project/issues",
          "search": "?files=foo%2Fbar.ts&resolved=false&id=my-project",
        }
      }
    >
      source_viewer.view_all_issues
    </ForwardRef(Link)>
  </div>
  <DeferredSpinner
    className="little-spacer-right"
  >
    <div
      className="flex-0 big-spacer-left"
    >
      <ButtonIcon
        aria-label="source_viewer.expand_all_lines"
        className="js-actions"
        onClick={[MockFunction]}
      >
        <ExpandSnippetIcon />
      </ButtonIcon>
    </div>
  </DeferredSpinner>
</div>
`;

exports[`should render correctly: no link to project 1`] = `
<div
  aria-label="foo/bar.ts"
  className="issue-source-viewer-header display-flex-row display-flex-space-between"
  role="separator"
>
  <div
    className="display-flex-center flex-1"
  >
    <div
      className="spacer-right"
    >
      <QualifierIcon
        qualifier="TRK"
      />
       
      <span>
        MyProject
      </span>
    </div>
    <div
      className="spacer-right"
    >
      <QualifierIcon
        qualifier="FIL"
      />
       
      <span>
        foo/
      </span>
      <span
        className="component-name-file"
      >
        bar.ts
      </span>
    </div>
    <div
      className="spacer-right"
    >
      <ClipboardIconButton
        aria-label="source_viewer.click_to_copy_filepath"
        className="button-link link-no-underline"
        copyValue="foo/bar.ts"
      />
    </div>
  </div>
  <div
    className="flex-0 big-spacer-left"
  >
    <ForwardRef(Link)
      to={
        {
          "hash": "",
          "pathname": "/project/issues",
          "search": "?files=foo%2Fbar.ts&resolved=false&id=my-project",
        }
      }
    >
      source_viewer.view_all_issues
    </ForwardRef(Link)>
  </div>
  <DeferredSpinner
    className="little-spacer-right"
  >
    <div
      className="flex-0 big-spacer-left"
    >
      <ButtonIcon
        aria-label="source_viewer.expand_all_lines"
        className="js-actions"
        onClick={[MockFunction]}
      >
        <ExpandSnippetIcon />
      </ButtonIcon>
    </div>
  </DeferredSpinner>
</div>
`;

exports[`should render correctly: no project name 1`] = `
<div
  aria-label="foo/bar.ts"
  className="issue-source-viewer-header display-flex-row display-flex-space-between"
  role="separator"
>
  <div
    className="display-flex-center flex-1"
  >
    <div
      className="spacer-right"
    >
      <QualifierIcon
        qualifier="FIL"
      />
       
      <span>
        foo/
      </span>
      <span
        className="component-name-file"
      >
        bar.ts
      </span>
    </div>
    <div
      className="spacer-right"
    >
      <ClipboardIconButton
        aria-label="source_viewer.click_to_copy_filepath"
        className="button-link link-no-underline"
        copyValue="foo/bar.ts"
      />
    </div>
  </div>
  <div
    className="flex-0 big-spacer-left"
  >
    <ForwardRef(Link)
      to={
        {
          "hash": "",
          "pathname": "/project/issues",
          "search": "?files=foo%2Fbar.ts&resolved=false&id=my-project",
        }
      }
    >
      source_viewer.view_all_issues
    </ForwardRef(Link)>
  </div>
  <DeferredSpinner
    className="little-spacer-right"
  >
    <div
      className="flex-0 big-spacer-left"
    >
      <ButtonIcon
        aria-label="source_viewer.expand_all_lines"
        className="js-actions"
        onClick={[MockFunction]}
      >
        <ExpandSnippetIcon />
      </ButtonIcon>
    </div>
  </DeferredSpinner>
</div>
`;

exports[`should render correctly: project root 1`] = `
<div
  aria-label="foo/bar.ts"
  className="issue-source-viewer-header display-flex-row display-flex-space-between"
  role="separator"
>
  <div
    className="display-flex-center flex-1"
  >
    <div
      className="spacer-right"
    >
      <a
        className="link-no-underline"
        href="/dashboard?id=my-project"
      >
        <QualifierIcon
          qualifier="TRK"
        />
         
        <span>
          MyProject
        </span>
      </a>
    </div>
  </div>
  <DeferredSpinner
    className="little-spacer-right"
  >
    <div
      className="flex-0 big-spacer-left"
    >
      <ButtonIcon
        aria-label="source_viewer.expand_all_lines"
        className="js-actions"
        onClick={[MockFunction]}
      >
        <ExpandSnippetIcon />
      </ButtonIcon>
    </div>
  </DeferredSpinner>
</div>
`;
